implementations/rust/ockam/ockam_entity/src/lib.rs

```diff
@@ -8,39 +8,86 @@
     unsafe_code,
     unused_import_braces,
     unused_qualifications,
-    // TODO re-enable warnings
+    // warnings
 )]
 
-use crate::EntityError;
+use std::sync::{Arc, Mutex};
+
+pub use change::*;
+pub use channel::*;
+pub use contact::*;
+pub use entity::*;
+pub use error::*;
+pub use identifiers::*;
+pub use key_attributes::*;
 use ockam_channel::SecureChannelVault;
-use ockam_core::lib::HashMap;
-use ockam_core::{Address, Result};
-use ockam_node::Context;
-use ockam_vault_core::{Hasher, KeyIdVault, SecretVault, Signer, Verifier};
-use ockam_vault_sync_core::VaultSync;
+use ockam_core::{lib::HashMap, Address, Message, Result};
+use ockam_node::{block_future, Context};
+use ockam_vault::{Hasher, KeyIdVault, SecretVault, Signer, Verifier};
+pub use profile_state::*;
+pub use traits::*;
+pub use worker::*;
+
+use crate::EntityError;
+
+pub struct ContextAddress {
+    ctx: Context,
+    address: Address,
+}
+
+pub type Handle = Arc<Mutex<ContextAddress>>;
```

Comment:
Not sure why we should use Arc and/or Mutex here

Reply:
This is to allow a Profile and an Entity to share the context. From call notes: Change to `clone`

Reply:
Was able to clean this up with your new_context suggestion:

```rust
pub struct Handle {
    ctx: Context,
    address: Address,
}

impl Clone for Handle {
    fn clone(&self) -> Self {
        block_future(&self.ctx.runtime(), async move {
            Handle {
                ctx: self
                    .ctx
                    .new_context(Address::random(0))
                    .await
                    .expect("new_context failed"),
                address: self.address.clone(),
            }
        })
    }
}```